napari/_qt/qt_about_keybindings.py

```diff
@@ -1,105 +1,80 @@
         )
-        scroll_widget.setContentsMargins(10, 10, 10, 10)
-        scroll_layout = QVBoxLayout()
-        scroll_layout.addWidget(self.active_label)
-        scroll_layout.addStretch(1)
-        scroll_widget.setLayout(scroll_layout)
-        self.setWidget(scroll_widget)
-
-        self.viewer.events.active_layer.connect(self.update_text)
-
-    def update_text(self, event):
-        keybindings_str = ''
+        current_layer = 'All active keybindings'
+        self.layerTypeComboBox.setCurrentText(current_layer)
+        # self.change_layer_type(current_layer)
+        layer_type_layout = QHBoxLayout()
+        layer_type_layout.setContentsMargins(10, 5, 0, 0)
+        layer_type_layout.addWidget(self.layerTypeComboBox)
+        layer_type_layout.addStretch(1)
+        layer_type_layout.setSpacing(0)
+        self.layout.addLayout(layer_type_layout)
+        self.layout.addWidget(self.textEditBox, 1)
+
+        self.viewer.events.active_layer.connect(self.update_active_layer)
+        self.update_active_layer(None)
+
+    def change_layer_type(self, text):
+        self.textEditBox.setHtml(self.keybindings_strs[text])
+
+    def update_active_layer(self, event):
+        text = ''
         # Add class and instance viewer keybindings
-        keybindings_str += get_keybindings_summary(self.viewer.class_keymap)
-        keybindings_str += get_keybindings_summary(self.viewer.keymap)
+        text += get_keybindings_summary(self.viewer.class_keymap)
+        text += get_keybindings_summary(self.viewer.keymap)
 
         layer = self.viewer.active_layer
         if layer is not None:
             # Add class and instance layer keybindings for the active layer
-            keybindings_str += get_keybindings_summary(layer.class_keymap)
-            keybindings_str += get_keybindings_summary(layer.keymap)
-        self.active_label.setText(keybindings_str)
-
-
-class QtLayerKeybindings(QWidget):
-    def __init__(self, layer):
-        super().__init__()
-
-        self.layout = QVBoxLayout()
+            text += get_keybindings_summary(layer.class_keymap)
+            text += get_keybindings_summary(layer.keymap)
 
-        # Add class keybindings for the layer
-        if len(layer.class_keymap) == 0:
-            keybindings_str = 'No keybindings'
-        else:
-            keybindings_str = get_keybindings_summary(layer.class_keymap)
-
-        layer_label = QLabel(keybindings_str)
-        layer_label.setTextInteractionFlags(Qt.TextSelectableByMouse)
-        # layer_label.setAlignment(Qt.AlignLeft)
-        layer_label.setSizePolicy(QSizePolicy.Expanding, QSizePolicy.Expanding)
-        self.setSizePolicy(QSizePolicy.Expanding, QSizePolicy.Expanding)
-        self.layout.addWidget(layer_label)
-        self.setLayout(self.layout)
+        # Do updates
+        self.keybindings_strs['All active keybindings'] = text
+        if self.layerTypeComboBox.currentText() == 'All active keybindings':
```

Comment:
what's this check for?

Reply:
If `All active keybindings` are displayed then when changing the active layer we need to update the actual display too. I've added a more informative comment